src/main/scala/mesosphere/marathon/core/instance/update/InstanceUpdater.scala

```diff
@@ -45,7 +45,16 @@ object InstanceUpdater extends StrictLogging {
             logger.info("all tasks of {} are terminal, requesting to expunge", updated.instanceId)
             InstanceUpdateEffect.Expunge(updated, events)
           } else {
-            InstanceUpdateEffect.Update(updated, oldState = Some(instance), events)
+            // If the updated task is Reserved, it means that the real task reached a Terminal state,
+            // which in turn means that the task managed to get up and running, which means that
+            // its persistent volume(s) had been created, and therefore they must never be destroyed/unreserved.
+            if (updatedTask.status.condition == Condition.Reserved) {
+              val suspendedState = Reservation.State.Suspended(timeout = None)
+              val suspended = updated.copy(reservation = updated.reservation.map(_.copy(state = suspendedState)))
```

Comment:
I would personally instead of such a deep object copying go more in this direction
instance.suspendReservation()
and move the rest of the code somewhere to the appropriate case classes

Btw. if ALL tasks are terminal, we can destroy the volume?
Because if (updated.tasksMap.values.forall(_.isTerminal)) -> then we do not suspend the reservation and from the comment you wrote I am not sure it is a bug or a correct behavior...

Reply:
Now, if all tasks are terminal, we do expunge an instance, but all tasks can be terminal here, only if the instance doesn't have PVs. Otherwise it is guaranteed that at least one of them is reserved.

Noted. I will avoid `.copy()`, when I will work on overhauling this part, but for now I will keep it as is in order to avoid testing one more time, because the fix has to be landed as soon as possible in order to include it into beta3

Reply:
BTW, please refer to `Task.update` method to the details on why it is guaranteed.

Reply:
oh ok, can you maybe add a note that all tasks being terminal cannot happen for tasks with PV? It was not obvious to me at the first glance.